RecoHGCal/TICL/plugins/HGCGraph.cc

```diff
@@ -125,12 +125,15 @@ void HGCGraph::makeAndConnectDoublets(const TICLLayerTiles &histo,
 
 bool HGCGraph::areTimeCompatible(int innerIdx,
                                  int outerIdx,
-                                 const edm::ValueMap<float> &layerClustersTime,
+                                 const edm::ValueMap<std::pair<float, float>> &layerClustersTime,
                                  float maxDeltaTime) {
-  float timeIn = layerClustersTime.get(innerIdx);
-  float timeOut = layerClustersTime.get(outerIdx);
+  float timeIn = layerClustersTime.get(innerIdx).first;
+  float timeInE = layerClustersTime.get(innerIdx).second;
+  float timeOut = layerClustersTime.get(outerIdx).first;
+  float timeOutE = layerClustersTime.get(outerIdx).second;
 
-  return (timeIn == -99 || timeOut == -99 || std::abs(timeIn - timeOut) < maxDeltaTime);
+  return (timeIn == -99 || timeOut == -99 ||
+          std::abs(timeIn - timeOut) < maxDeltaTime * sqrt(timeInE * timeInE + timeOutE * timeOutE));
```

Comment:
Would it make sense to also check that the Time Error is meaningful, i.e. it is >0?
I'm not sure if we can have a reasonable time but not a meaningful error.

Reply:
No, should not be the case:
- the error for recHits is sampled from a positive function [between 0.02 and 5.5 ns] for hits with time (and set to -1 for recHits in the scintillator and those without time)
- only hits with time error >= 0 are contributing to layer clusters 
- all time and timeError estimates in cascade are from a weighted mean, so starting with positive input errors only a positive error on the mean is allowed
- I don't see any way the timeError can be < 0, except when time = -99 (-99, -1 are the default values returned for no time). But this is already taken into account

If either than this timeError is <= 0, it is for a bug somewhere else in the chain.

Can I suggest to leave this as it is? (to help spotting the bug in case?)

Reply:
Sure, thanks for checking!